src/main/scala/com/ironcorelabs/IronSdkAdvancedSync.scala

```diff
@@ -3,26 +3,22 @@ package com.ironcorelabs.scala.sdk
 import cats.effect.Sync
 import cats.implicits._
 import scodec.bits.ByteVector
+import com.ironcorelabs.{sdk => jsdk}
 
-case class IronSdkAdvancedSync[F[_]](deviceContext: DeviceContext)(implicit syncF: Sync[F]) extends IronSdkAdvanced[F] {
-  val underlying = deviceContext.toJava.map(d => com.ironcorelabs.sdk.IronSdk.initialize(d).advanced)
+case class IronSdkAdvancedSync[F[_]](underlying: jsdk.IronSdkAdvanced)(implicit syncF: Sync[F])
+    extends IronSdkAdvanced[F] {
 
   def documentEncryptUnmanaged(data: ByteVector, options: DocumentEncryptOpts): F[DocumentEncryptUnmanagedResult] =
     for {
       javaOpts <- options.toJava
-      result <- underlying.map(
-        _.documentEncryptUnmanaged(
-          data.toArray,
-          javaOpts
-        )
-      )
+      result = underlying.documentEncryptUnmanaged(data.toArray, javaOpts)
     } yield DocumentEncryptUnmanagedResult(result)
 
   def documentDecryptUnmanaged(
     encryptedData: EncryptedData,
     encryptedDeks: EncryptedDeks
   ): F[DocumentDecryptUnmanagedResult] =
-    underlying
-      .map(_.documentDecryptUnmanaged(encryptedData.underlyingBytes, encryptedDeks.underlyingBytes))
+    syncF
+      .delay(underlying.documentDecryptUnmanaged(encryptedData.underlyingBytes, encryptedDeks.underlyingBytes))
```

Comment:
I don't understand this `.delay` call here, but my understanding of `Sync` might be a little Rusty. I'm mostly wondering why only the decrypt calls have this.

Reply:
It's there because it could throw. And it was an oversight that it was only on the decrypts, I've now added them basically everywhere after a suggestion from Colt.